pkg/agent/sysadvisor/plugin/inference/modelresultfetcher/borwein/borwein.go

```diff
@@ -245,38 +258,40 @@ func (bmrf *BorweinModelResultFetcher) getInferenceRequestForPods(requestContain
 			continue
 		}
 
-		if req.PodRequestEntries[containerInfo.PodUID] == nil {
-			req.PodRequestEntries[containerInfo.PodUID] = &borweininfsvc.ContainerRequestEntries{
-				ContainerFeatureValues: make(map[string]*borweininfsvc.FeatureValues),
-			}
-		}
-
 		unionFeatureValues := &borweininfsvc.FeatureValues{
 			Values: make([]string, 0, len(req.FeatureNames)),
 		}
 
 		unionFeatureValues.Values = append(unionFeatureValues.Values, nodeFeatureValues...)
 
 		for _, containerFeatureName := range bmrf.containerFeatureNames {
-			if getContainerFeatureValue == nil {
-				return nil, fmt.Errorf("nil getContainerFeatureValue")
-			}
-
 			containerFeatureValue, err := getContainerFeatureValue(containerInfo.PodUID,
 				containerInfo.ContainerName, containerFeatureName,
 				metaServer, metaReader)
-
+			// Just continue, ignore this container.
+			// And let getContainerFeatureValue decide which feature is allowed to return default value.
 			if err != nil {
-				return nil, fmt.Errorf("getContainerFeatureValue for pod: %s/%s, container: %s failed",
```

Comment:
after discussion, 

1. getContainerFeatureValue should be refined to return nil err and zero value when input feature isn't a key feature
2. if err != nil, we just return err here to avoid meaningless request to inference sever